packages/components/src/internal/components/picklist/actions.ts

```diff
@@ -0,0 +1,233 @@
+import { PICKLIST, PRIVATE_PICKLIST_CATEGORY, PUBLIC_PICKLIST_CATEGORY } from '../domainproperties/list/constants';
+import { insertRows, InsertRowsResponse } from '../../query/api';
+import { SchemaQuery } from '../../../public/SchemaQuery';
+import { getSelected, getSelectedData } from '../../actions';
+import { List } from 'immutable';
+import { saveDomain } from '../domainproperties/actions';
+import { QueryModel } from '../../../public/QueryModel/QueryModel';
+import { Actions } from '../../../public/QueryModel/withQueryModels';
+import { User } from '../base/models/User';
+import { PicklistModel } from './models';
+import { flattenValuesFromRow } from '../../../public/QueryModel/utils';
+import { buildURL } from '../../url/AppURL';
+import { fetchListDesign } from '../domainproperties/list/actions';
+import { resolveErrorMessage } from '../../util/messaging';
+
+interface CreatePicklistResponse {
+    domainId: number,
+    name: string
+}
+
+export function setPicklistDefaultView(name: string): Promise<CreatePicklistResponse> {
+    return new Promise((resolve, reject) => {
+        let jsonData = {
+            schemaName: "lists",
+            queryName: name,
+            views: [{
+                columns:[
+                    {fieldKey: "SampleID/Name"},
+                    {fieldKey: "SampleID/LabelColor"},
+                    {fieldKey: "SampleID/SampleSet"},
+                    {fieldKey: "SampleID/StoredAmount"},
+                    {fieldKey: "SampleID/Units"},
+                    {fieldKey: "SampleID/freezeThawCount"},
+                    {fieldKey: "SampleID/StorageStatus"},
+                    {fieldKey: "SampleID/checkedOutBy"},
+                    {fieldKey: "SampleID/Created"},
+                    {fieldKey: "SampleID/CreatedBy"},
+                    {fieldKey: "SampleID/StorageLocation"},
+                    {fieldKey: "SampleID/StorageRow"},
+                    {fieldKey: "SampleID/StorageCol"},
+                    {fieldKey: "SampleID/isAliquot"},
+                ]
+            }]
+        };
+        return Ajax.request({
+            url: buildURL('query', 'saveQueryViews.api'),
+            method: 'POST',
+            jsonData,
+            success: Utils.getCallbackWrapper(response => {
+                resolve(response);
+            }),
+            failure: Utils.getCallbackWrapper(response => {
+                console.error(response);
+                reject("There was a problem creating the default view for the picklist. " + resolveErrorMessage(response));
+            }),
+        });
+    });
+}
+
+export function createPicklist(name: string, description: string, shared: boolean) : Promise<CreatePicklistResponse> {
+    return new Promise((resolve, reject) => {
+        Domain.create({
+            domainDesign: {
+                name: name,
+                fields: [
+                    {
+                        name: 'SampleID',
+                        rangeURI: 'int',
```

Comment:
do we also want to set the `conceptURI` to `SAMPLE_TYPE_CONCEPT_URI` for this field so that the data type shows up as "Sample" instead of lookup. Oh wait, the lookup here is to the inventory.SampleItems not exp.Materials....hmmm maybe not then.